File: apps/api/test/fixtures/multi-lang/payment.go
// @ts-nocheck
package payment

import (
	"fmt"
	"time"
)

type PaymentConfig struct {
	ApiKey  string
	Timeout time.Duration
}

type PaymentService struct {
	config PaymentConfig
}

func NewPaymentService(config PaymentConfig) *PaymentService {
	return &PaymentService{config: config}
}

func (s *PaymentService) ProcessPayment(amount float64, currency string) (bool, error) {
	fmt.Printf("Processing %f %s\n", amount, currency)
	return true, nil
}

func (s *PaymentService) Refund(transactionID string) error {
	fmt.Printf("Refunding %s\n", transactionID)
	return nil
}

func CalculateTax(amount float64, rate float64) float64 {
	return amount * rate
}

type PremiumPaymentService struct {
	PaymentService
}

func (p *PremiumPaymentService) ApplyDiscount(amount float64, discount float64) float64 {
	return amount * (1 - discount)
}

File: apps/api/test/fixtures/multi-lang/payment.rs
// @ts-nocheck
pub struct PaymentConfig {
    pub api_key: String,
    pub timeout: u64,
}

pub trait PaymentProcessor {
    fn process_payment(&self, amount: f64, currency: &str) -> Result<bool, String>;
}

pub struct PaymentService {
    config: PaymentConfig,
}

impl PaymentService {
    pub fn new(config: PaymentConfig) -> Self {
        PaymentService { config }
    }

    pub fn process_payment(&self, amount: f64, currency: &str) -> Result<bool, String> {
        println!("Processing {} {}", amount, currency);
        Ok(true)
    }

    pub fn refund(&self, transaction_id: &str) {
        println!("Refunding {}", transaction_id);
    }
}

pub fn calculate_tax(amount: f64, rate: f64) -> f64 {
    amount * rate
}

pub struct PremiumPaymentService {
    inner: PaymentService,
}

impl PremiumPaymentService {
    pub fn apply_discount(&self, amount: f64, discount: f64) -> f64 {
        amount * (1.0 - discount)
    }
}
